### PR TITLE
.Net: Rename SKFunction factory methods

### PR DESCRIPTION
**Pull request by stephentoub**
The only difference between these two methods should be how the method information is provided, whether it's a single delegate or a combination of a MethodInfo and a target object. This PR thus renames both methods to just be overloads of the same Create method, accepting the same parameters other than the exact ones which specify the method.  It also cleans up the implementation in various ways, removing code duplication, removing extraneous allocation, fixing XML comments, etc.

Fixes https://github.com/microsoft/semantic-kernel/issues/3165

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

**Review by markwallace-microsoft — APPROVED** [review]

**Comment by markwallace-microsoft** [comment]
@stephentoub There are some merge conflicts, can you resolve this and we can get this merged

**Comment by stephentoub** [comment]
I've addressed the conflicts.

File: dotnet/src/SemanticKernel.Core/Functions/SKFunction.cs [hunk 1]
```diff
@@ -11,13 +12,56 @@
...
+    /// <param name="pluginName">The optional name of the plug-in associated with this method.</param>
+    /// <param name="functionName">Optional function name. If null, it will default to one derived from the method represented by <paramref name="method"/>.</param>
+    /// <param name="description">Optional description of the method. If null, it will default to one derived from the method represented by <paramref name="method"/>, if possible (e.g. via a <see cref="DescriptionAttribute"/> on the method).</param>
+    /// <param name="parameters">Optional parameter descriptions. If null, it will default to one derived from the method represented by <paramref name="method"/>.</param>
+    /// <param name="returnParameter">Optional return parameter description. If null, it will default to one derived from the method represented by <paramref name="method"/>.</param>
+    /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
+    /// <returns>The created <see cref="ISKFunction"/> wrapper for <paramref name="method"/>.</returns>
+    public static ISKFunction Create(
+        Delegate method,
```
Comment (SergeyMenshykh) [inline dotnet/src/SemanticKernel.Core/Functions/SKFunction.cs:32]: nit: should the parameter name be delegate or function to be aligned with the `functionName` parameter?

**Review by markwallace-microsoft — APPROVED** [review]

File: dotnet/src/SemanticKernel.Core/Functions/SKFunction.cs (hunk 1 above)
Comment (stephentoub) [reply dotnet/src/SemanticKernel.Core/Functions/SKFunction.cs:32]: The "method" here is about the .NET method being wrapped.
The "functionName" parameter refers to the name to use for ISKFunction.

**Review by SergeyMenshykh — APPROVED** [review]

**Pull request merged**